### ISSUE TITLE
saveLayout is not serializable/deserializable

### ISSUE THREAD

**Issue by GordonSmith** [phosphorjs/phosphor#309]
saveLayout includes entire widget objects and not just the layout:
https://github.com/phosphorjs/phosphor/blob/23b9d075ebc5b73ab148b6ebfc20af97f85714c4/packages/widgets/src/docklayout.ts#L1201-L1216

I was expecting the following capability:
* User opens app
* User messes with layout
* App saves layout to DB
* User closes app
...
* User opens app
* Previous layout exists in DB and its layout is restored 
* User messes with layout
* App saves layout to DB
* User closes app
...rinse and repeat...

Something more like:
```typescript
    createConfig(): DockLayout.ITabAreaConfig {
      let widgets = this.tabBar.titles
          .filter(title => title.owner && title.owner.createConfig)
          .map(title => title.owner.createConfig());
      let currentIndex = this.tabBar.currentIndex;
      return { type: 'tab-area', widgets, currentIndex };
    }
```

I suspect the current implementation is more for a quick undo/redo type scenario?

**Comment by sccolbert** [comment]
Duplicate of #280 

TL;DR You need to build your own serialization/deserialization on top of this mechanism. It wouldn't be right for Phosphor to make those decisions for you.
